docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
 # Developer Guide
+## Contents
+* [Acknowledgements](#acknowledgements)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [Manager Component](#manager-component)
+    * [Profile Component](#profile-component)
+    * [TextUI Component](#textui-component)
+    * [Storage Component](#storage-component)
+    * [Parser Component](#parser-component)
+    * [Command Component](#command-component)
+* Implementation
+    * [Add a session](#add-a-session)
+    * [Remove a session](#remove-a-session)
+    * [View a session](#view-a-session)
+    * [List all sessions](#list-sessions)
+    * [Settle a session](#settle-a-session)
+    * [Add an activity](#add-an-activity)
+    * [Remove an activity](#remove-an-activity)
+    * [View an activity](#view-an-activity)
+    * [List all activities](#list-activities)
+    * [Add a group](#add-a-group)
+    * [Remove a group](#remove-a-group)
+    * [View a group](#view-a-group)
+    * [List all groups](#list-groups)
+* [Project Scope](#product-scope)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+
 
 ## Acknowledgements
+* Address Book (Level-3) - Provide samples of User Guide(UG) and Developer Guide(DG)
+
+## Design
+### Architecture
+![Application Diagram Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ArchitectureDiagram.drawio.png)
```

Comment:
Is the sequence diagram too detailed? It is hard to see what is being written in the diagram unless it is zoomed into.
What I see at default zoom level:
![image](https://user-images.githubusercontent.com/69501969/160956338-e57f8454-b978-454f-813b-6d6869f506d7.png)

Perhaps you can consider splitting it up into different parts (e.g. 1 diagram for loading the file, 1 diagram for processing user input).